### PR TITLE
C++: Catch more returns of stack-allocated memory

### PR DESCRIPTION
**Pull request by MathiasVP**
@andersfugmann reminded me that there's an evil family of functions that return stack-allocated memory, and that our `cpp/return-stack-allocated-memory` query isn't aware of these. This PR fixes this problem.

### CODE REVIEW

File: cpp/ql/src/Likely Bugs/Memory Management/ReturnStackAllocatedMemory.ql [hunk 1]
```diff
@@ -27,16 +27,26 @@ class ReturnStackAllocatedMemoryConfig extends MustFlowConfiguration {
...
+        var = source and
+        var.getAstVariable() instanceof StackVariable and
+        // Pointer-to-member types aren't properly handled in the dbscheme.
+        not var.getResultType() instanceof PointerToMemberType
+      )
+      or
+      // `source` is an instruction that represents the return value of a
+      // function that is known to return stack-allocated memory.
+      exists(Call call |
+        call.getTarget().hasGlobalName(["alloca", "strdupa", "strndupa"]) and
```
Comment (andersfugmann) [inline cpp/ql/src/Likely Bugs/Memory Management/ReturnStackAllocatedMemory.ql:47]: mscv has `_alloca`. Maybe also add that.

Comment (MathiasVP) [reply cpp/ql/src/Likely Bugs/Memory Management/ReturnStackAllocatedMemory.ql:47]: Good call. I also see that there's https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/malloca?view=msvc-170. I will add these

Comment (MathiasVP) [reply cpp/ql/src/Likely Bugs/Memory Management/ReturnStackAllocatedMemory.ql:47]: (I see that `_malloca ` could potentially be safe if it's guaranteed to allocate memory above the `_ALLOCA_S_THRESHOLD` threshold, but I don't think this should exclude it from the query.)

Comment (MathiasVP) [reply cpp/ql/src/Likely Bugs/Memory Management/ReturnStackAllocatedMemory.ql]: Fixed in https://github.com/github/codeql/commit/bae7e10e463c7e62e80f60daa5b5508f3eddf076

**Review by jketema — APPROVED** [review]
LGTM. I do like to see a complete DCA experiment.

**Comment by MathiasVP** [comment]
Nothing exciting on DCA. Merging!

**Pull request merged**